lib/puppet/provider/exec/powershell.rb

```diff
@@ -27,28 +27,30 @@
         }
   EOT
 
-  POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
-  Currently, the PowerShell module has reduced v1 functionality on this agent
-  due to one or more of the following conditions:
+  unless defined?(POWERSHELL_MODULE_UPGRADE_MSG)
+    POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
```

Comment:
You might be able to use `||=` instead of `unless defined?`

Reply:
@Iristyle I was unaware of the existence of this operator. I modified the commit to use this operator. Thanks for pointing this out.